File: go/fixture/go-mini/internal/handlers/status.go
package handlers

import (
	"fmt"
	"net/http"

	"example.com/go-mini/internal/repository"
)

// Status returns the status handler.
func Status(store repository.Store) http.HandlerFunc {
	return func(w http.ResponseWriter, r *http.Request) {
		devices, err := store.List(r.Context())
		if err != nil {
			http.Error(w, "store unavailable", http.StatusServiceUnavailable)
			return
		}
		var ready, degraded, down, other int
		for _, d := range devices {
			switch d.Status {
			case "READY":
				ready++
			case "DEGRADED":
				degraded++
			case "DOWN":
				down++
			default:
				other++
			}
		}
		w.Header().Set("Content-Type", "text/plain; charset=utf-8")
		_, _ = fmt.Fprintf(w, "devices=%d ready=%d degraded=%d down=%d other=%d\n",
			len(devices), ready, degraded, down, other)
	}
}

File: py/fixture/py-mini/internal/handlers/status.py
"""The fleet status line."""

from http import HTTPStatus

from internal.handlers.mux import HandlerFunc, Request, Response, text
from internal.repository.store import Store


def status(store: Store) -> HandlerFunc:
    """Return the status handler."""

    def handle(_req: Request) -> Response:
        try:
            devices = store.list_devices()
        except OSError:
            return text(HTTPStatus.SERVICE_UNAVAILABLE, "store unavailable")
        ready = degraded = down = other = 0
        for d in devices:
            match d.status:
                case "READY":
                    ready += 1
                case "DEGRADED":
                    degraded += 1
                case "DOWN":
                    down += 1
                case _:
                    other += 1
        return text(
            HTTPStatus.OK,
            f"devices={len(devices)} ready={ready} degraded={degraded} down={down} other={other}",
        )

    return handle
